clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputMapper.java

```diff
@@ -29,22 +30,26 @@ protected AbstractVeniceRecordReader<BytesWritable, KafkaInputMapperValue> getRe
   }
 
   @Override
-  protected Optional<AbstractVeniceFilter<KafkaInputMapperValue>> getFilter(final VeniceProperties props) {
+  protected FilterChain<KafkaInputMapperValue> getFilterChain(final VeniceProperties props) {
+    FilterChain<KafkaInputMapperValue> filterChain = new FilterChain<>();
     long ttlInHours = props.getLong(VenicePushJob.REPUSH_TTL_IN_HOURS, VenicePushJob.NOT_SET);
     if (ttlInHours != VenicePushJob.NOT_SET) {
-      VeniceKafkaInputTTLFilter filter = new VeniceKafkaInputTTLFilter(props);
-      return Optional.of(filter);
+      try {
+        filterChain.add(new VeniceKafkaInputTTLFilter(props));
```

Comment:
Why not use the same constructor to pass a list of filters all the time? And we could remove `add` function from `FilterChain` class.

Reply:
Imagining we have multiple filters need to be added to the chain, and we have to add them accordingly inside some `if`/`switch` statement, I think the `add` function will become handy.